homeassistant/components/fan/__init__.py

```diff
@@ -47,37 +45,23 @@
 SUPPORT_DIRECTION = 4
 SUPPORT_PRESET_MODE = 8
 
-SERVICE_SET_SPEED = "set_speed"
 SERVICE_INCREASE_SPEED = "increase_speed"
 SERVICE_DECREASE_SPEED = "decrease_speed"
 SERVICE_OSCILLATE = "oscillate"
 SERVICE_SET_DIRECTION = "set_direction"
 SERVICE_SET_PERCENTAGE = "set_percentage"
 SERVICE_SET_PRESET_MODE = "set_preset_mode"
 
-SPEED_OFF = "off"
-SPEED_LOW = "low"
-SPEED_MEDIUM = "medium"
-SPEED_HIGH = "high"
-
 DIRECTION_FORWARD = "forward"
 DIRECTION_REVERSE = "reverse"
 
-ATTR_SPEED = "speed"
 ATTR_PERCENTAGE = "percentage"
 ATTR_PERCENTAGE_STEP = "percentage_step"
-ATTR_SPEED_LIST = "speed_list"
 ATTR_OSCILLATING = "oscillating"
 ATTR_DIRECTION = "direction"
 ATTR_PRESET_MODE = "preset_mode"
 ATTR_PRESET_MODES = "preset_modes"
 
-_NOT_SPEED_OFF = "off"
-
-OFF_SPEED_VALUES = [SPEED_OFF, None]
-
-LEGACY_SPEED_LIST = [SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
-
 
 class NoValidSpeedsError(ValueError):
```

Comment:
Can we remove the exceptions for invalid speed?

Reply:
86fb7cff08ea0adcf54f0929de1e89787460f7ea